aws/resource_aws_api_gateway_deployment_test.go

```diff
@@ -69,6 +112,32 @@ func testAccCheckAWSAPIGatewayDeploymentExists(n string, res *apigateway.Deploym
 	}
 }
 
+func testAccCheckAWSAPIGatewayDeploymentStageExists(deploymentName string, stageName string, res *apigateway.Stage) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		conn := testAccProvider.Meta().(*AWSClient).apigateway
+
+		deploymentResource, ok := s.RootModule().Resources[deploymentName]
+		if !ok {
+			return fmt.Errorf("Deployment not found: %s", deploymentName)
+		}
+
+		restApiId := aws.String(deploymentResource.Primary.Attributes["rest_api_id"])
+
+		req := &apigateway.GetStageInput{
+			StageName: &stageName,
```

Comment:
Nitpick: the stage name can be fetched from attributes as well (`deploymentResource.Primary.Attributes["stage_name"]`) instead of a parameter passed in from the acceptance test